Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/audit/COREX12RealtimeTransforms.java

```diff
@@ -0,0 +1,342 @@
+        auditMsg.getAuditSourceIdentification().add(auditSource);
+
+        result.setAuditMessage(auditMsg);
+        result.setDirection(direction);
+        result.setInterface(_interface);
+
+        LOG.debug("end transformRequestToAuditMsg() -- NHIN");
+        return result;
+    }
+
+    /**
+     * 
+     * @param response
+     * @param assertion
+     * @param target
+     * @param direction
+     * @param _interface
+     * @param isRequesting
+     * @return LogEventRequestType
+     */
+    public LogEventRequestType transformResponseToAuditMsg(COREEnvelopeRealTimeResponse response, AssertionType assertion,
+            NhinTargetSystemType target, String direction, String _interface, boolean isRequesting) {
+        LogEventRequestType result = null;
+        AuditMessageType auditMsg = null;
+
+        if (response == null) {
+            LOG.error("Response Object was null");
+            return null;
+        }
+
+        if (assertion == null) {
+            LOG.error("Assertion was null");
+            return null;
+        }
+
+        // check to see that the required fields are not null
+        boolean missingReqFields = areRequiredCOREX12RealtimeFiledsNull(response, assertion);
+        if (missingReqFields) {
+            LOG.error("One or more required fields was missing");
+            return null;
+        }
+
+        result = new LogEventRequestType();
+        auditMsg = new AuditMessageType();
+
+        // Create EventIdentification
+        CodedValueType eventID = getCodedValueTypeForXDRResponse();
+        EventIdentificationType oEventIdentificationType = getEventIdentificationType(eventID);
+        auditMsg.setEventIdentification(oEventIdentificationType);
+
+        AuditMessageType.ActiveParticipant participant = getActiveParticipant(assertion.getUserInfo());
+        auditMsg.getActiveParticipant().add(participant);
+
+        /* Assign ParticipationObjectIdentification */
+        ParticipantObjectIdentificationType participantObject = getParticipantObjectIdentificationType(response.getPayloadID());
+        auditMsg.getParticipantObjectIdentification().add(participantObject);
+
+        /* Create the AuditSourceIdentifierType object */
+        String communityId = oXDR.getMessageCommunityIdFromRequest(assertion, target, direction, _interface);
+        AuditSourceIdentificationType auditSource = getAuditSourceIdentificationType(communityId);
+        auditMsg.getAuditSourceIdentification().add(auditSource);
+
+        result.setAuditMessage(auditMsg);
+        result.setDirection(direction);
+        result.setInterface(_interface);
+
+        LOG.debug("end transformResponseToAuditMsg() -- NHIN");
+
+        return result;
+    }
+
+    private boolean areRequiredCOREX12RealtimeFiledsNull(COREEnvelopeRealTimeResponse msg, AssertionType assertion) {
+        try {
+
+            if (assertion == null) {
```

Comment:
This seems redundant.  You already check if assertion and message are null before.  I would delete one of those groups of checks.